src/economy/economy.cpp

```diff
@@ -3812,6 +3812,10 @@ void resolve_constructions(sys::state& state) {
 				break;
 			}
 		}
+
+		if(state.world.province_building_construction_get_nation(c) == state.local_player_nation && state.cheat_data.instant_construction)
```

Comment:
similarly, I would rather see a "finish buildings" cheat that fills up the purchased construction goods once in the command rather than an additional check here